src/tests/components/LocationDetails.test.js

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import renderer from "react-test-renderer";
+import LocationDetails from "../../components/LocationDetails";
+
+describe("LocationDetails", () => {
+  const validProps = {
+    city: "Test city",
+    country: "Test country",
+  };
+
+  it("renders correctly", () => {
+    const rendered = renderer.create(
+      <LocationDetails city={validProps.city} country={validProps.country} />
+    );
+    expect(rendered).toMatchSnapshot();
+  });
+
+  it("renders correct city and location", () => {
+    render(
+      <LocationDetails city={validProps.city} country={validProps.country} />
+    );
+
+    const locationDetails = screen.getByText("Test city, Test country");
+    expect(locationDetails).toBeInTheDocument();
```

Comment:
Its common convention to leave a clear line before the assertions start in each test to help with readability, should add one in here